client/web/src/user/settings/auth/ExternalAccountsSignIn.tsx

```diff
@@ -65,35 +79,63 @@ const getNormalizedAccount = (
     if (account && accountExternalData) {
         switch (kind) {
-            case 'GITHUB':
-                {
-                    const githubExternalData = accountExternalData as GitHubExternalData
-                    normalizedAccount = {
-                        ...normalizedAccount,
-                        external: {
-                            id: account.id,
-                            // map github fields
-                            userName: githubExternalData.name,
-                            userLogin: githubExternalData.login,
-                            userUrl: githubExternalData.html_url,
-                        },
-                    }
+            case 'github': {
+                const githubExternalData = accountExternalData as GitHubExternalData
+                normalizedAccount = {
+                    ...normalizedAccount,
+                    external: {
+                        id: account.id,
+                        // map GitHub fields
+                        userName: githubExternalData.name,
+                        userLogin: githubExternalData.login,
+                        userUrl: githubExternalData.html_url,
+                    },
                 }
+            }
                 break
-            case 'GITLAB':
-                {
-                    const gitlabExternalData = accountExternalData as GitLabExternalData
-                    normalizedAccount = {
-                        ...normalizedAccount,
-                        external: {
-                            id: account.id,
-                            // map gitlab fields
-                            userName: gitlabExternalData.name,
-                            userLogin: gitlabExternalData.username,
-                            userUrl: gitlabExternalData.web_url,
-                        },
+            case 'gitlab': {
+                const gitlabExternalData = accountExternalData as GitLabExternalData
+                normalizedAccount = {
+                    ...normalizedAccount,
+                    external: {
+                        id: account.id,
+                        // map gitlab fields
+                        userName: gitlabExternalData.name,
+                        userLogin: gitlabExternalData.username,
+                        userUrl: gitlabExternalData.web_url,
+                    },
+                }
+            }
+                break
+            case 'saml': {
+                // The data stored in user_external_account has this structure: “Values”[“<attribute names for email or username>"] > Values[0] > “Value”: string
+                let email = ''
+                let username = ''
+
+                const samlExternalData = accountExternalData as SamlExternalData
+                if (samlExternalData.Values) {
+                    const entries = Object.entries(samlExternalData.Values)
+                    for (const [name, att] of entries) {
```

Comment:
You don't have to do `Object.entries(samlExternalData.Values)`, you can just directly iterate over it:

```suggestion
                    for (const [name, att] of samlExternalData.Values) {
```

This preserves type information. `Object.entries` returns `att` as an `any` type so you have no type information past that point and would have to cast it again

Reply:
Since you can call `Object.entries()` on it means it should be iterable. You can read about `Object.entries()` here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries

But the part I want to highlight is "This is the same as iterating with a for...in loop". So I'm wondering where you are seeing that error? Can you perhaps share a screenshot?